dhall-nix/src/Dhall/Nix.hs

```diff
@@ -488,6 +495,12 @@ dhallToNix e = loop (Dhall.Core.normalize e)
         a' <- loop a
         b' <- loop b
         return (Fix (NBinary NUpdate a' b'))
+    loop (Field (Union kts) k) = do
+        let e0 = do
+                k' <- Dhall.Map.keys kts
+                return (k', Nothing)
+        let e2 = Fix (NSym k)
+        return (Fix (NAbs (ParamSet e0 False Nothing) e2))
```

Comment:
If I understand correctly, this compiles `< Foo >.Foo` to `{ Foo }: Foo`, but I believe the result should instead be: `x: { Foo }: Foo x`

Reply:
`<Foo>.Foo` compiling to `{Foo}: Foo` seems correct to me - what is that `x` about?

Reply:
Oh, you're right.  Never mind, my thoughts were confused

Reply:
Now I remember what led to my confusion.  The issue here is that you need to check if the alternative is empty when compiling the union to Nix.

For example:

```nix
dhallToNix "< Foo >.Foo" = { Foo }: Foo

dhallToNix "< Foo : {} >.Foo {=}" = { Foo }: Foo {}

dhallToNix "< Foo : {} >.Foo" = x: { Foo }: Foo x
```

Carefully note the difference between the 1st and 3rd examples.  If the type of the alternative is present then you need to turn it into an anonymous function.

Reply:
Ah, so it's a case of `< .. >.x` being partially applied when the alternative does take data

Reply:
Ok, I believe my latest commit resolves this.